internal/backends/collection.go

```diff
@@ -199,6 +201,28 @@ func (cc *collectionContract) Explain(ctx context.Context, params *ExplainParams
 	return res, err
 }
 
+// CollectionStatsParams represents the parameters of Collection.Stats method.
+type CollectionStatsParams struct{}
+
+// CollectionStatsResult represents the results of Collection.Stats method.
+type CollectionStatsResult struct {
+	CountObjects   int64
+	CountIndexes   int64
+	SizeTotal      int64
+	SizeIndexes    int64
+	SizeCollection int64
+}
+
+// Stats returns statistics about the collection.
+func (cc *collectionContract) Stats(ctx context.Context, params *CollectionStatsParams) (*CollectionStatsResult, error) {
+	defer observability.FuncCall(ctx)()
+
+	res, err := cc.c.Stats(ctx, params)
+	checkError(err, ErrorCodeCollectionDoesNotExist)
```

Comment:
For `collStats` command, indeed it returns zero-filled struct.
However, aggregation `$collStats` stage returns command error, and existing compat test would fail if error isn't returned.

https://github.com/FerretDB/FerretDB/pull/3295/files#diff-b26835ea59cb934e35c55f22543373b2954d6a2f81ed7901d94d2b5e9ee0ade1R388

Reply:
Then, could we do the opposite and return `ErrorCodeDatabaseDoesNotExist` in `Database.Stats`? That would make both `Stats` method consistent

Reply:
That would work, let me do that :+1: